### ISSUE TITLE
Enhancement: Add sym links

### ISSUE THREAD

**Issue by nicoabie** [fnogatz/swivm#17]
I needed to run the following commands to access swi as a library.

ln -s ~/.swivm/versions/8.0.3/lib/swipl/lib/x86_64-linux/libswipl.so /usr/lib/libswipl.so
ln -s ~/.swivm/versions/8.0.3/lib/swipl/lib/x86_64-linux/libswipl.so.8 /usr/lib/libswipl.so.8

I think maybe swivm can be responsible for mantaining those always pointing to the current version in use.

Does it make sense?

**Comment by fnogatz** [comment]
I am not really familiar with accessing SWI as a library using the `.so` files. Maybe you give me a pointer to the corresponding section in the SWI-Prolog manual?

My first impression is, that instead of calling
```sh
ln -s ~/.swivm/versions/8.0.3/lib/swipl/lib/x86_64-linux/libswipl.so /usr/lib/libswipl.so
ln -s ~/.swivm/versions/8.0.3/lib/swipl/lib/x86_64-linux/libswipl.so.8 /usr/lib/libswipl.so.8
```
we should aim for something like
```sh
ln -s ~/.swivm/alias/current/lib/swipl/lib/x86_64-linux/libswipl.so /usr/lib/libswipl.so
ln -s ~/.swivm/alias/current/lib/swipl/lib/x86_64-linux/libswipl.so.8 /usr/lib/libswipl.so.8
```
and handle a new alias `current` separately by `swivm`. This way, the symlinks have to be created only once, at installation of `swivm`, instead of on each installation of a new version of SWI-Prolog. I am not yet sure how to best integrate this, as swivm currently does not require root permissions, which however are needed to create a symlink in `/usr/lib`. Asking for root-permissions is always a sensible thing and possible cause for major failures :)

**Comment by nicoabie** [comment]
@fnogatz I'm not familiar either but while using a python package called pyswip which access swi as a library, I had to create those.

You are right, asking for root permissions is a sensible thing, maybe after setting a current version swivm could tell the user that in order to access it as a lib the symlinks you wrote should be created.

I believe that when installing swi using apt-get the /usr/lib directory get those files.

**Comment by fnogatz** [comment]
I just added `swivm lib [<version>]` which sets the symbolic links in `/usr/lib`. Note that it's currently not very flexible and does not do any magic to check for different systems and paths (`/usr/lib` or something different?) as well as for the `libswipl.so` paths. If any future user experiences problems, just open a new issue :)

The workflow is to `swivm install <version>` and then call `swivm lib`, or `swivm lib <version>` if you have multiple versions installed. It is never called automatically, as it requires sudo permissions (unlike all other operations of swivm).
